manifests/sssd/client.pp

```diff
@@ -38,40 +43,49 @@
 # @param min_id
 #   The lowest user ID that SSSD should recognize from the remote server
 #
-# @author Trevor Vaughan <tvaughan@onyxpoint.com>
+# @author https://github.com/simp/pupmod-simp-simp/graphs/contributors
 #
 class simp::sssd::client (
   Boolean $ldap_domain       = simplib::lookup('simp_options::ldap', { 'default_value' => false }),
-  Boolean $local_domain      = true,
-  Boolean $autofs            = true,
-  Boolean $sudo              = true,
-  Boolean $ssh               = true,
+  Boolean $local_domain,     #data in module
+  Boolean $autofs            = true, #deprecated
+  Boolean $sudo              = true, #deprecated
+  Boolean $ssh               = true, #deprecated
   Boolean $enumerate_users   = false,
   Boolean $cache_credentials = true,
   Integer $min_id            = 500
 ){
 
   simplib::module_metadata::assert($module_name, { 'blacklist' => ['Windows'] })
 
-  if $ldap_domain or $local_domain {
-    include 'sssd'
-    include 'sssd::service::nss'
-    include 'sssd::service::pam'
+  # Don't attemt to setup sssd in el6 or 7 if a local or ldap domain is not defined.
 
-    if $autofs { include 'sssd::service::autofs' }
-    if $sudo { include 'sssd::service::sudo' }
-    if $ssh { include 'sssd::service::ssh' }
+  if $local_domain or $ldap_domain or versioncmp($facts['os']['release']['major'], '8') >= 0 {
+
+    include 'sssd'
 
     if $local_domain {
+
+      if $facts['os']['release']['major'] <= '6' {
```

Comment:
This should be a `versioncmp`

Reply:
I realize I was inconsistent and use versioncmp in one instance and in the other I didn't.   Bt your comments in one say why are you using it and in the next one say why are you NOT using it.  I would think it does not need to be used.  Is that OK?

Reply:
Oh, it's that the first is successfully handled by the `$local_domain` and `$ldap_domain` variables combined with the module data so I don't think we need the EL8 comparison at all.

In terms of `versioncmp()` I just started using it whenever I was comparing versions because I kept finding that string comparisons didn't always do what I expected.

Reply:
I changed them to versioncmp.  I el8 you don't need a domain set up  and it will set up the files domain automatically.  It recommended that you do this because it is quicker lookup.  It is a valid configuration where in el7 and el 6 it was not.